src/Api/APIRest.php

```diff
@@ -335,13 +335,20 @@ public function call()
                     if (!isset($this->parameters['input'])) {
                         $this->messageBadArrayError();
                     }
-                    // if id is passed by query string, add it into input parameter
-                    $input = (array) ($this->parameters['input']);
-                    if (
-                        ($id > 0 || $id == 0 && $itemtype == "Entity")
```

Comment:
Maybe removing the casting of `$this->parameters['input']` and checking it is an object before trying to define its ID is a simplier solution.

```php
// if id is passed by query string, add it into input parameter
if (
    is_object($this->parameters['input'])
    && ($id > 0 || $id == 0 && $itemtype == "Entity")
    && !property_exists($this->parameters['input'], 'id')
) {
    $this->parameters['input']->id = $id;
}
```